daemon/images/service.go

```diff
@@ -116,29 +119,24 @@ func (i *ImageService) Children(_ context.Context, id image.ID) ([]image.ID, err
 // CreateLayer creates a filesystem layer for a container.
 // called from create.go
 // TODO: accept an opt struct instead of container?
-func (i *ImageService) CreateLayer(container *container.Container, initFunc layer.MountInit) (layer.RWLayer, error) {
+func (i *ImageService) CreateLayer(ctx context.Context, container *container.Container, _ string, _ *ocispec.Platform, setupInit func(string) error) error {
```

Comment:
Related to the "abstraction". From this it really looks like the abstraction no longer really "works". We're now have to change the signature to effectively be for "snapshot", and more and more stubbing for graphdrivers. Perhaps diversifying starts to be a better options 🤔 (which would also allow for more descriptive naming, clearer on intent)